stdlib/src/builtin/string_literal.mojo

```diff
@@ -129,6 +129,60 @@ struct StringLiteral(
         """
         return not self == rhs
 
+    @always_inline("nodebug")
+    fn __lt__(self, rhs: StringLiteral) -> Bool:
+        """Compare this StringLiteral to the RHS using LT comparison.
+
+        Args:
+            rhs: The other StringLiteral to compare against.
+
+        Returns:
+            True if this StringLiteral is strictly less than the RHS StringLiteral and False otherwise.
+        """
+        var len1 = len(self)
+        var len2 = len(rhs)
+
+        if len1 < len2:
```

Comment:
Suggestion: you can consolidate the duplication between this and the implementation in String by implementing the comparison on StringRef

Reply:
Do you mean something like `return StringRef(self) < StringRef(rhs)` as `__lt__` for StringLiteral (and `return self._strref_dangerous() < rhs._strref_dangerous()` for String), etc.?

I didn't even consider StringRef (forgot it existed), and I tried to match `__lt__` with their respective `__eq__`-methods. Right now String, StringLiteral, and StringRef does `__eq__` in three separate ways, so I thought there was a reason for keeping them apart(??). Would it still be safe to use `memory.memcmp` in the StringRef `__lt__` implementation (considering it currently looks like StringLiteral has to use it's own local `_memcmp`, and StringRef's `__eq__` also avoids it), or should StringRef's `__lt__` be done like its `__eq__`?

What do you think of:
```mojo
    @always_inline("nodebug")
    fn __lt__(self, rhs: StringRef) -> Bool:
        """Compare this StringRef to the RHS using LT comparison.

        Args:
            rhs: The other StringRef to compare against.

        Returns:
            True if this StringRef is strictly less than the RHS StringRef and False otherwise.
        """
        var len1 = len(self)
        var len2 = len(rhs)

        if len1 < len2:
            return memcmp(self.data, rhs.data, len1) <= 0
        else:
            return memcmp(self.data, rhs.data, len2) < 0
```